# Analyzing Gun Deaths in the US 2012-2014

In [16]:
import csv

# I am importing the CSV file containing the gun 
# deaths data and creating a list of lists, in order
# to separate the rows into their own list.

f = open('guns.csv', 'r')
data = list(csv.reader(f))
data[:5]

[['',
  'year',
  'month',
  'intent',
  'police',
  'sex',
  'age',
  'race',
  'hispanic',
  'place',
  'education'],
 ['1',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '34',
  'Asian/Pacific Islander',
  '100',
  'Home',
  '4'],
 ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'],
 ['3',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '60',
  'White',
  '100',
  'Other specified',
  '4'],
 ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]

In [17]:
# I must first remove the header row, and assign
# it to its own variable in order to analyze the
# data.

headers = data[0]
data = data[1:]

headers

['',
 'year',
 'month',
 'intent',
 'police',
 'sex',
 'age',
 'race',
 'hispanic',
 'place',
 'education']

In [18]:
data[:5]

[['1',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '34',
  'Asian/Pacific Islander',
  '100',
  'Home',
  '4'],
 ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'],
 ['3',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '60',
  'White',
  '100',
  'Other specified',
  '4'],
 ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'],
 ['5',
  '2012',
  '02',
  'Suicide',
  '0',
  'M',
  '31',
  'White',
  '100',
  'Other specified',
  '2']]

## Gun Deaths Per Year

In [25]:
# I will extract the year column from the data in
# order to calculate the number of gun deaths each
# year.

years = [row[1] for row in data]

year_counts = {}

for year in years:
    if year in year_counts:
        year_counts[year] += 1
    else:
        year_counts[year] = 1

year_counts

{'2012': 33563, '2013': 33636, '2014': 33599}

## Gun Deaths Per Month from 2012-2014

In [27]:
import datetime as dt

# Using the datetime object, the gun deaths per
# month can be calculated and stored into a
# dictionary easily.

dates = [dt.datetime(year=int(row[1]), 
         month=int(row[2]),
         day=1) for row in data]
date_counts = {}

for date in dates:
    if date in date_counts:
        date_counts[date] += 1
    else:
        date_counts[date] = 1

date_counts

{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

## Gun Deaths by Sex and Race

In [29]:
sex_counts = {}
race_counts = {}

# Following a similar procedure as above, I will
# calculate the statistics for gun deaths by sex
# and race.

for row in data:
    sex = row[5]
    race = row[7]
    if sex in sex_counts:
        sex_counts[sex] += 1
    else:
        sex_counts[sex] = 1
    if race in race_counts:
        race_counts[race] += 1
    else:
        race_counts[race] = 1

sex_counts

{'F': 14449, 'M': 86349}

In [30]:
race_counts

{'Asian/Pacific Islander': 1326,
 'Black': 23296,
 'Hispanic': 9022,
 'Native American/Native Alaskan': 917,
 'White': 66237}

When filtered by sex and race the data shows to heavily favor certain values. For gender it favors males with a count of 86,349 deaths compared to 14,449 females. For race it seems favors whites with a count of 66,237 deaths. With blacks being a distant second at 23,296 deaths. To really understand the number of gun deaths per race, we have to compare it to the proportion of each race in the US population.

Right now we have the raw counts of gun deaths per race, we need to express the data as gun deaths rate per 100,000.

In [32]:
f = open('census.csv', 'r')
census = list(csv.reader(f))
census

[['Id',
  'Year',
  'Id',
  'Sex',
  'Id',
  'Hispanic Origin',
  'Id',
  'Id2',
  'Geography',
  'Total',
  'Race Alone - White',
  'Race Alone - Hispanic',
  'Race Alone - Black or African American',
  'Race Alone - American Indian and Alaska Native',
  'Race Alone - Asian',
  'Race Alone - Native Hawaiian and Other Pacific Islander',
  'Two or More Races'],
 ['cen42010',
  'April 1, 2010 Census',
  'totsex',
  'Both Sexes',
  'tothisp',
  'Total',
  '0100000US',
  '',
  'United States',
  '308745538',
  '197318956',
  '44618105',
  '40250635',
  '3739506',
  '15159516',
  '674625',
  '6984195']]

## Gun Death Rate per Race per 100000

In [44]:
# Total populations per race in the US
mapping = {'Asian/Pacific Islander': int(census[1][14])+int(census[1][15]),
           'Black': int(census[1][12]),
           'Hispanic': int(census[1][11]),
           'Native American/Native Alaskan': int(census[1][13]),
           'White': int(census[1][10])
           }
# Dividing the gun deaths per race values by the total
# population values in the US and multiplying the
# product by 100,000 to get the gun deaths rate per
# 100,000.

race_per_hundredk = {}
for race in race_counts:
    race_per_hundredk[race] = (race_counts[race] / mapping[race]) * 100000

race_per_hundredk

{'Asian/Pacific Islander': 8.374309664161762,
 'Black': 57.8773477735196,
 'Hispanic': 20.220491210910907,
 'Native American/Native Alaskan': 24.521955573811088,
 'White': 33.56849303419181}

The values above show a different story than the statistics from just the raw values. When you look at the data as gun death rate per race per 100000 it shows to be much more likely for blacks to be the victim of a gun death at a rate of 57.88/100000 due to the higher total population of whites vs blacks in the US.

## Homicide Rates per Race per 100000

In [52]:
intents = [row[3] for row in data]
races = [row[7] for row in data]

homicide_race_counts = {}

# Homicide Gun Death totals separated by Race

for i, race in enumerate(races):
    if intents[i] == "Homicide":
        if race in homicide_race_counts:
            homicide_race_counts[race] += 1
        else:
            homicide_race_counts[race] = 1
# Homicide Gun Death rate per 100,000 separated by Race

homicide_race_totals = homicide_race_counts
print(homicide_race_totals)

for race in homicide_race_counts:
    homicide_race_counts[race] = (homicide_race_counts[race] / mapping[race]) * 100000

print('\n')
homicide_race_counts


{'Asian/Pacific Islander': 559, 'White': 9147, 'Hispanic': 5634, 'Black': 19510, 'Native American/Native Alaskan': 326}




{'Asian/Pacific Islander': 3.530346230970155,
 'Black': 48.471284987180944,
 'Hispanic': 12.627161104219914,
 'Native American/Native Alaskan': 8.717729026240365,
 'White': 4.6356417981453335}

The data shows that it is <strong>much</strong> more likely for the victim of a gun death to be someone who is black at 48.47/100000.